test/common/index.js

```diff
@@ -672,11 +672,34 @@ function invalidArgTypeHelper(input) {
   return ` Received type ${typeof input} (${inspected})`;
 }
 
+async function collectStream(readable, callback) {
+  let data = '';
+  try {
+    readable.setEncoding('utf8');
+    for await (const chunk of readable)
+      data += chunk;
+    if (typeof callback === 'function') callback(null, data);
```

Comment:
I would wrap this callback in a nextTick to avoid any error being catched by the outer try-catch

Reply:
Done (the code is now in `common/streams.js`).